### PR TITLE
Support single-sided periods in TELx rewards calculator

### PR DESCRIPTION
**Pull request by leopashov**
## Summary

Adds a single-sided fallback to the TELx rewards calculation pipeline so that periods where one side of the pool sees zero global fee growth no longer abort with "No fee growth in one or both assets during the period." Instead, those periods rank LPs by their fees on the growing side alone.

Previously `main()` called `getVolumeWeightedAveragePriceScaled`, which threw inside `getFeeGrowthGlobalsPeriodDelta` whenever either `feeGrowthDelta0` or `feeGrowthDelta1` was zero. A genuinely one-sided week (e.g. a pool where only currency0 generated swap fees) therefore couldn't be finalised at all. The rewards logic now distinguishes two cases up-front via a new `VwapResult` discriminator:

- `{ kind: "vwap", priceScaled }` -> both sides grew; convert the non-denominator side using the VWAP implied by the fee ratio.
- `{ kind: "singleSided", growingIsCurrency0 }` -> one side flat; rank LPs directly by their weighted fees on the growing side. The absent side's price would be a common scalar and cancels out in the downstream share ratio, so the ranking is provably identical to what a VWAP-based allocation would produce if a price existed.

## Key changes

`backend/calculators/TELxRewardsCalculator.ts`

- `deriveVwapResult(delta0, delta1, denomIs0)` - single decision point for whether a period is two-sided (trades ocurred in both directions) or one-sided (trades only happened in one direction). Returns a tagged `VwapResult` and contains the VWAP arithmetic previously inside `getVolumeWeightedAveragePriceScaled`.
- `populateTotalFeesCommonDenominator` - new singleSided branch. When `vwapResult.kind === "singleSided"`, each LP's total becomes `periodFeesCurrency0Weighted` or `periodFeesCurrency1Weighted` depending on `growingIsCurrency0`. LPs out of range on the growing side (zero accrual) get a zero total and receive no weight.
- `getFeeGrowthGlobalsPeriodDelta` - removed the throw on zero deltas; the caller now decides how to interpret them.
- `getVolumeWeightedAveragePriceScaled` - deleted (functionality covered by `deriveVwapResult` + `getFeeGrowthGlobalsPeriodDelta`).
- `main()` - now calls `getFeeGrowthGlobalsPeriodDelta` followed by `deriveVwapResult`, and writes the resulting `VwapResult` into `CheckpointData.vwapResult` so each checkpoint records which branch it took.
- `parseCLIArgs` - added --rerun flag. When passed, output is written to `<name>-<period>.rerun.json` instead of overwriting the canonical checkpoint, supporting byte-for-byte regression testing against historical runs.
- Updated docstrings on `populateTotalFeesCommonDenominator` and the `main()` usage header

## Tests

`backend/test/TELxRewardsCalculator.test.ts` - 13 unit tests covering:

- A1–A6: `deriveVwapResult` - both sides growing (both denomIs0 orientations), only currency0 grew, only currency1 grew, both flat, and an end-to-end pipeline test showing that a fully flat period returns an empty reward map via the pre existing `calculateRewardDistribution` guard.
- B1–B5: `populateTotalFeesCommonDenominator` - both VWAP orientations, both singleSided orientations, and the out-of-range LP case (zero accrual → zero ranking weight).
- C1–C2: Distribution equivalence - same fixture run through both branches yields identical rewards. This is the algebraic justification for using the growing side alone as the ranking metric in the singleSided case.

`backend/test/TELxRewardsEquivalence.test.ts` - end-to-end regression test that re-runs period 34 on Polygon ETH-TEL under the new pipeline and compares the produced checkpoint to the committed canonical one (polygon-ETH-TEL-34.json), ignoring only the newly added `vwapResult` field. Passes, confirming the two-sided code path is byte-identical to the previous implementation.

## Checkpoint artefacts

- `backend/checkpoints/polygon-ETH-TEL-34.rerun.json` - committed rerun fixture used by the equivalence test.
- `backend/checkpoints/polygon-ETH-TEL-35.json`, `backend/checkpoints/base-ETH-TEL-35.json` - new period 35 outputs produced under the updated pipeline (worked before update.
- `backend/checkpoints/polygon-USDC-EMXN-35.json` - new period 35 output using single sided logic branch

## Notes / out of scope

- The human-readable XLSX exporter (backend/telxHumanReadable.ts) destructures a fixed set of fields from the checkpoint JSON and silently ignores vwapResult, so existing reports are unaffected, and script functionality remains. Same for safeTxArrayBuilder.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2772,22 +2784,22 @@ yallist@^3.0.2:
...
+  integrity sha512-y11nGElTIV+CT3Zv9t7VKl+Q3hTQoT9a1Qzezhhl6Rp21gJ/IVTW7Z3y9EWXhuUBC2Shnf+DX0antecpAwSP8w==
+
 yargs-parser@^21.1.1:
   version "21.1.1"
   resolved "https://registry.npmjs.org/yargs-parser/-/yargs-parser-21.1.1.tgz"
   integrity sha512-tVpsJW7DdjecAiFpbIB1e3qxIQsE6NoPc5/eTdrbbIC4h0LVsWhnoa3g+m2HclBIujHzsxZ4VJVA+GUuc2/LBw==
 
-yargs-unparser@2.0.0:
+yargs-unparser@^2.0.0:
   version "2.0.0"
```
Comment (Huwonk) [inline yarn.lock:2791]: I'd usually pin package versions for consistency between builds/deploys/environments.  Doing ^ means "this or greater" and could have unpredictable side effects.  Luckily the lock files prevent it, but if the lock files ever got corrupted or someone rebuilt them from the package.json, it could break something.  And usually it'd be something really hard to detect.  I'm going to approve anyway, since these particular changes have limited impact, just more of a note / caveat.

**Review by Huwonk — APPROVED** [review]
Good stuff, Leo.  Looks clean.  Tests pass. I left a note regarding package pinning, and I recommend in the future to always use a pinned npm package and let it get updated via dependabot with socket.dev safety checks to a pinned/static version, but I found no issues with the current state at all.  Really more feedback than a blocker of any sort.

**Comment by leopashov** [comment]
@Huwonk Reverted package changes

File: backend/test/TELxRewardsCalculator.test.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+function makeLPMap(
+  entries: Array<[Address, bigint, bigint]>,
+): Map<Address, LPData> {
+  const m = new Map<Address, LPData>();
+  for (const [addr, f0, f1] of entries) m.set(addr, makeLP(f0, f1));
+  return m;
+}
+
+describe("deriveVwapResult", () => {
```
Comment (chasebrownn) [inline backend/test/TELxRewardsCalculator.test.ts:32]: There doesnt seem to be a test for equal deltas. Something like `deriveVwapResult(100n, 100n, true)` should yield `priceScaled === PRECISION (1:1 price)`. Low-risk but a good boundary case.

File: backend/test/TELxRewardsCalculator.test.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      lpData,
+      { kind: "singleSided", growingIsCurrency0: true },
+      false,
+    );
+    expect(result.get(LP_A)?.totalFeesCommonDenominatorWeighted).to.equal(0n);
+    expect(result.get(LP_B)?.totalFeesCommonDenominatorWeighted).to.equal(40n);
+  });
+});
+
+describe("distribution equivalence when one side is globally flat", () => {
```
Comment (chasebrownn) [inline backend/test/TELxRewardsCalculator.test.ts:184]: C1 and C2 both use `growingIsCurrency0: true`. An equivalence test for the `growingIsCurrency0: false` side (zero currency0 fees pool-wide) might be a good idea. Not sure if this is a probable case or if its something to worry about. But in terms of test coverage this looks like a possible gap.

**Review by chasebrownn — APPROVED** [review]

File: backend/test/TELxRewardsCalculator.test.ts (hunk 3 above)
Comment (leopashov) [reply backend/test/TELxRewardsCalculator.test.ts:184]: Added in final commit

File: backend/test/TELxRewardsCalculator.test.ts (hunk 2 above)
Comment (leopashov) [reply backend/test/TELxRewardsCalculator.test.ts:32]: Added in final commit
